### PR TITLE
Virtual Console for Pokeyellow

### PR DESCRIPTION
**Pull request by vulcandth**
Nintendo Virtual Console Patch Support for Pokeyellow. Please reference pret/pokecrystal#813 pret/pokecrystal#882 pret/pokegold#74 pret/pokered#351 for more info.

Nothing too notably different for pokeyellow compared to the other repo's. There is one thing i'd like looked at further. `[FPA 031801 Begin]` and `[FPA 031801 Begin2]` appear to refer to the `YellowIntroScene`'s. I didn't want to hardcode the values in the template.. so I gave them pseudo labels in the `pokeyellow.constants.asm` file as below. I don't know how I feel about this.. and there is probably a better way to do this... maybe something more dynamic. Perhaps I am also just doing something wrong. Let me know!

### CODE REVIEW

File: vc/pokeyellow.constants.asm [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+; These are all the asm constants needed to make the blue_vc patch.
+
+vc_const: MACRO
+x = \1
+	PRINTLN "00:{04x:x} \1" ; same format as rgblink's .sym file
+ENDM
+
+; [FPA 031801 Begin1]
+YellowIntroScene14_Index = $e
```
Comment (Rangi42) [inline vc/pokeyellow.constants.asm:11]: This looks like the right time to introduce a file with actual `const_def`/`const` sequences for the Yellow-only Pikachu-related data, even if they're just placeholder constants like some of the ones in move_animation_constants.asm.

File: vc/pokeyellow.patch.template [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+;34 6F 26 00
+;Mem Write: pc32 = 0xf9e51 addr = 0xc635 value = 0x33   
+[FPA 031801 End1]                
+Mode = 3                         
+Type = 1                         
+Address = {hex @}                
+ConditionType = 0                
+ConditionValueA = {dws_ wYellowIntroCurrentScene wYellowIntroSceneTimer}     
+ConditionValueB = {dws_ ==                       ==                    }     
+ConditionValueC = {dws_ YellowIntroScene14_Index 0x33                  }
```
Comment (Rangi42) [inline vc/pokeyellow.patch.template:232]: Is this an index into something, or just some arbitrary timer value?

File: vc/pokeyellow.patch.template [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+;17 --13     
+;No151 Mega Punch            
+[FPA 001 Begin]                       
+Mode = 3                              
+Type = 0                              
+Address = {hex @}                     
+MotionBEnable0 = 3                    
+MotionBlur0 = 11                       
+ConditionType = 11                     
+ConditionValueA = {dws_ wStringBuffer wStringBuffer+1 wStringBuffer+2 wStringBuffer+3 wStringBuffer+5 wAnimationID 00 wStringBuffer wStringBuffer+1 wStringBuffer+2 wStringBuffer+3 wAnimationID 00 wStringBuffer wStringBuffer+1 wStringBuffer+2 wStringBuffer+3 wAnimationID}
```
Comment (Rangi42) [inline vc/pokeyellow.patch.template:303]: Double-space around `00`/`||`/`00` please.

File: vc/pokeyellow.patch.template (hunk 2 above)
Comment (vulcandth) [reply vc/pokeyellow.patch.template]: From what I understand this is an arbitrary timer value.. From what I can tell; the scenes use this value sometimes to determine when to transition to other sequences. 0x33 doesn't like up with anything particular that I am able to tell.

**Pull request merged**